cluster/src/main/java/org/apache/iotdb/cluster/qp/executor/QueryMetadataExecutor.java

```diff
@@ -177,6 +185,123 @@ public String processMetadataInStringQuery()
     return combineMetadataInStringList(metadataList);
   }
 
+  public Metadata processMetadataQuery()
+      throws InterruptedException, ProcessorException, PathErrorException {
+    Set<String> groupIdSet = router.getAllGroupId();
+
+    Metadata[] metadatas = new Metadata[groupIdSet.size()];
+    List<SingleQPTask> taskList = new ArrayList<>();
+    for (String groupId : groupIdSet) {
+      QueryMetadataRequest request = new QueryMetadataRequest(groupId,
+          readMetadataConsistencyLevel);
+      SingleQPTask task = new SingleQPTask(false, request);
+      taskList.add(task);
+
+      LOGGER.debug("Execute query metadata statement for group {}.", groupId);
+      /** Check if the plan can be executed locally. **/
+      if (canHandleQueryByGroupId(groupId)) {
+        LOGGER.debug("Execute query metadata statement locally for group {}.", groupId);
```

Comment:
Since current node can handle this request, why you provide a private function called asyncQueryMetadataLocally. It seems that this function is quite similar to QueryMetadataAsyncProcessor's handleRequest funtion.
You can send a request to itself?

Reply:
Because even though we can handle this request locally, we still need to used readIndex function to get result, and this process is asynchronously.